### PR TITLE
Varying resolution support

### PR DESCRIPTION
**Pull request by Ridleh**
in dooley 2 fashion this comes with limitations
- adds support to allow the user to view the game in resolutions outside 1920x1080p
- resolutions smaller than native resolution while fullscreened leads to the game being in full screen but in the topleft corner of the screen, not centered
-UI components that are not scaled (e.g replace button text) may be too big or too small depending on resolution
-props to the original students once again for doing all of the heavy lifting :D

### CODE REVIEW

**Comment by Ridleh** [comment]
please do not merge until #483 is resolved

**Comment by fluffy-hamster** [comment]
Also be aware we should not merge this until we have figured out a way to update appData config files.

**Comment by Ridleh** [comment]
please merge AFTER #483 is merged

**Comment by fluffy-hamster** [comment]
has this been tested with larger resolutions ?

**Comment by Ridleh** [comment]
> 
> 
> has this been tested with larger resolutions ?

no. i only have a single 1080p monitor :/

**Comment by skeet70** [comment]
I tested this with a 1440p monitor and didn't have any issues. I can test it with a 4k one later today too. I'm on Linux though and haven't ever actually had issues with it scaling even before this, so I can't tell for sure that it fixes the original problem since it wasn't appearing for me.

File: Client/src/main/java/org/projectcardboard/client/controller/GraphicalUserInterface.java [hunk 1]
```diff
@@ -58,15 +60,16 @@ public void setBackgroundMusic(Media media) {
...
+        boolean setFullScreen = Boolean.parseBoolean(Config.getInstance().getProperty("FULLSCREEN"));
+        stage.setFullScreen(setFullScreen);
         stage.setResizable(true); // note that resizing the window does not currently 'redraw' the window.
 
         stage.setFullScreenExitHint(KeyboardShortcutConstants.EXIT_FULLSCREEN_HELP_MSG);
         stage.setFullScreenExitKeyCombination(KeyboardShortcutConstants.EXIT_FULLSCREEN);
 
         stage.show();
         stage.setOnCloseRequest(event -> Client.getInstance().close());
+
```
Comment (skeet70) [inline Client/src/main/java/org/projectcardboard/client/controller/GraphicalUserInterface.java]:
```suggestion
```

File: Client/src/main/java/org/projectcardboard/client/models/gui/UIConstants.java [hunk 2]
```diff
@@ -21,11 +22,13 @@
                     Color.rgb(44, 33, 129), CornerRadii.EMPTY, Insets.EMPTY
             )
     );
+    private static final double DEFAULT_SCREEN_WIDTH = Double.parseDouble(Config.getInstance().getProperty("SCREEN_WIDTH"));
+    private static final double DEFAULT_SCREEN_HEIGHT = Double.parseDouble(Config.getInstance().getProperty("SCREEN_HEIGHT"));
```
Comment (skeet70) [inline Client/src/main/java/org/projectcardboard/client/models/gui/UIConstants.java]:
```suggestion
    private static final double CONFIG_SCREEN_WIDTH = Double.parseDouble(Config.getInstance().getProperty("SCREEN_WIDTH"));
    private static final double CONFIG_SCREEN_HEIGHT = Double.parseDouble(Config.getInstance().getProperty("SCREEN_HEIGHT"));
```

File: Client/src/main/java/org/projectcardboard/client/models/gui/UIConstants.java [hunk 3]
```diff
@@ -21,11 +22,13 @@
...
     private static final double DEFAULT_SCENE_WIDTH = 3636;
     private static final double DEFAULT_SCENE_HEIGHT = 2045;
     public static final double SCALE = Math.min(
-            Screen.getPrimary().getBounds().getWidth() / DEFAULT_SCENE_WIDTH,
-            Screen.getPrimary().getBounds().getHeight() / DEFAULT_SCENE_HEIGHT
+            (DEFAULT_SCREEN_WIDTH / DEFAULT_SCENE_WIDTH),
+            (DEFAULT_SCREEN_HEIGHT / DEFAULT_SCENE_HEIGHT)
```
Comment (skeet70) [inline Client/src/main/java/org/projectcardboard/client/models/gui/UIConstants.java]:
```suggestion
            (CONFIG_SCREEN_WIDTH / DEFAULT_SCENE_WIDTH),
            (CONFIG_SCREEN_HEIGHT / DEFAULT_SCENE_HEIGHT)
```

File: Client/src/main/java/org/projectcardboard/client/models/gui/UIConstants.java [hunk 4]
```diff
@@ -39,6 +42,9 @@
     
     static{
         try {
+            System.out.println(Screen.getPrimary().getBounds().getWidth());
```
Comment (skeet70) [inline Client/src/main/java/org/projectcardboard/client/models/gui/UIConstants.java]: What are these three lines for? If we want to keep them, combine them into a single print statement with something like "Native screen resolution: width x height"

File: Server/src/main/resources/config.properties [hunk 5]
```diff
@@ -4,9 +4,14 @@
...
+SERVER_IP=3.123.26.96
+SERVER_URI=wss://mechaz.org
+PORT=8888
+SERVER_NAME=Server
+API_HOST=mechaz.org
+
+#UI Related Variables
+SCREEN_WIDTH=1920
+SCREEN_HEIGHT=1080
+FULLSCREEN=true
```
Comment (skeet70) [inline Server/src/main/resources/config.properties]: Did your autoformatter switch this file to no spaces between identifiers?

Comment (Ridleh) [reply Server/src/main/resources/config.properties]: in #493 we have a method that appends to the user's config file the missing config variables if theres any. But the check to see if the user is missing some variables only works if we remove the spaces between identifiers.

File: Client/src/main/java/org/projectcardboard/client/models/gui/UIConstants.java (hunk 4 above)
Comment (Ridleh) [reply Client/src/main/java/org/projectcardboard/client/models/gui/UIConstants.java]: thats just me being sloppy in removing all of my print statements :/

**Pull request merged**